src/abstractdataframe/iteration.jl

```diff
@@ -159,15 +160,17 @@ julia> sum.(eachcol(df))
 """
 eachcol(df::AbstractDataFrame) = DataFrameColumns(df)
 
-Base.size(itr::DataFrameColumns) = (size(parent(itr), 2),)
-Base.IndexStyle(::Type{<:DataFrameColumns}) = Base.IndexLinear()
-
-@inline function Base.getindex(itr::DataFrameColumns, j::Int)
-    @boundscheck checkbounds(itr, j)
-    @inbounds parent(itr)[!, j]
-end
-
-Base.getindex(itr::DataFrameColumns, j::Symbol) = parent(itr)[!, j]
+Base.length(itr::DataFrameColumns) = size(parent(itr), 2)
+Base.eltype(::Type{<:DataFrameColumns}) = AbstractVector
```

Comment:
This definition is not needed as it is already present by default:
```
julia> Base.IteratorEltype(typeof(eachcol(df)))
Base.HasEltype()

julia> Base.IteratorEltype(eachcol(df))
Base.HasEltype()
```

Reply:
Ah indeed. Though, should we set it to `EltypeUnknown()` because we don't know the concrete type of columns? `?Base.IteratorEltype` says:
>   This trait is generally used to select between algorithms that pre-allocate a specific type of result, and algorithms that pick a result type based on the types of yielded values.

Not saying we should do it, but it's worth thinking about it.